### PR TITLE
implements integration for Nuki Opener

### PR DESCRIPTION
**Pull request by sschum**

### CODE REVIEW

File: lock.go [hunk 1]
```diff
@@ -3,17 +3,21 @@ package nuki
...
 	"context"
 	"fmt"
+	"github.com/tarent/go-nuki/communication"
 	"github.com/tarent/go-nuki/communication/command"
 )
 
-// ReadLockState will request the current lock state for the connected and paired nuki device and return the result.
-func (c *Client) ReadLockState(ctx context.Context) (command.KeyturnerStatesCommand, error) {
+// ReadLockerState will request the current locker state for the connected and paired nuki device and return the result.
+func (c *Client) ReadLockerState(ctx context.Context) (command.KeyturnerStatesCommand, error) {
```
Comment (sschum) [inline lock.go:11]: Mabe i should also generalize the state such like the config (for consistency)

File: general.go [hunk 2]
```diff
@@ -104,3 +122,68 @@ func (c *Client) Close() error {
 
 	return nil
 }
+
+// PerformAction will request the connected and paired nuki opener to perform the given command.
```
Comment (memoos) [inline general.go:126]: As far as I see this is not only for opener but also for smart lock...

Comment (sschum) [reply general.go:126]: yeah thats correct :innocent:

File: communication/error.go [hunk 3]
```diff
@@ -111,19 +118,30 @@ func (c Command) Error() error {
...
+		}
+		return K_ERROR_LOW_VOLTAGE
 	case 0x4A:
 		return K_ERROR_MOTOR_POWER_FAILURE
 	case 0x4B:
-		return K_ERROR_CLUTCH_POWER_FAILURE
+		if deviceType == DeviceTypeSmartLock {
+			return K_ERROR_CLUTCH_POWER_FAILURE
+		}
+		return K_ERROR_RECORDING_TIMEOUT
```
[the remark this answers is not in the document]
Comment (sschum) [reply communication/error.go:138]: you are right :+1:

File: lock.go (hunk 1 above)
Comment (sschum) [reply lock.go:11]: for the lock action we cant do that because there is a slightly different action id used for open (0x03) and unlocking (0x01)

**Review by memoos — APPROVED** [review]

**Pull request merged**